Shopify/ruby-lsp
# Add code lens for test methods

### Motivation

Closes #584 
A continuation of #598 

See updated gif for demo.

### Implementation

- Extended the functionality of the `CodeLens` request
- Added an instance variable `@modifier`
    - Visitor visits every `SyntaxTree::VCall` to check if it is an access modifier, if so updates `@modifier`
    - Visits every `SyntaxTree::Command` to check if it is a `public` access modifier, if so temporarily updates `@modifer` for that one method, then reverts the modifier back to what it was previously. (ex. `public def test_method(); end`)
- Visits every method definition to check if it is a public method to starts with "test"
   - If so, creates a new code lens for the method (setting the `path` to the specific line number the test begins and setting the `command` to the appropriate terminal command to run the test.
 - Refactored `add_code_lens` a bit to make the parameters easier to read and understand.

### Automated Tests

Updated automated testing.

### Manual Tests

See manual test instructions in https://github.com/Shopify/vscode-ruby-lsp/pull/506. 
- Run individual tests instead of running all class tests.
- Look at demo for example
